### PR TITLE
[Feature] Support SegNeXt in MMSegmentation 2.0

### PR DESCRIPTION
**Pull request by MengzhangLI**
## Motivation

Support SegNeXt in MMSeg 1.x branch.

0.x PR: https://github.com/open-mmlab/mmsegmentation/pull/2600

WIP: README.md needs to be refined...

### CODE REVIEW

File: configs/segnext/segnext_mscan-t_1xb16-adamw-160k_ade20k-512x512.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        keep_ratio=True),
+    dict(type='RandomCrop', crop_size=crop_size, cat_max_ratio=0.75),
+    dict(type='RandomFlip', prob=0.5),
+    dict(type='PhotoMetricDistortion'),
+    dict(type='PackSegInputs')
+]
+test_pipeline = [
+    dict(type='LoadImageFromFile'),
+    dict(type='Resize', scale=(2048, 512), keep_ratio=True),
+    dict(type='ResizeToMultiple', size_divisor=32),
```
Comment (xiexinch) [inline configs/segnext/segnext_mscan-t_1xb16-adamw-160k_ade20k-512x512.py:76]: Might resize at data_preprocessor by setting `test_cfg` to data_preprocessor

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -42,16 +42,16 @@ repos:
...
+#  - repo: local
+#    hooks:
+#      - id: update-model-index
+#        name: update-model-index
+#        description: Collect model information and update model-index.yml
+#        entry: .dev/md2yml.py
+#        additional_dependencies: [mmengine, lxml, opencv-python]
+#        language: python
+#        files: ^configs/.*\.md$
+#        require_serial: true
```
Comment (xiexinch) [inline .pre-commit-config.yaml:54]: Might restore changes to this file.

File: configs/segnext/segnext.yml [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    Title: 'SegNeXt: Rethinking Convolutional Attention Design for Semantic Segmentation'
+  README: configs/segnext/README.md
+  Code:
+    URL: https://github.com/open-mmlab/mmsegmentation/blob/v0.31.0/mmseg/models/backbones/mscan.py#L328
+    Version: v0.31.0
+  Converted From:
+    Code: https://github.com/visual-attention-network/segnext
+Models:
+- Name: ''
+  In Collection: SegNeXt
```
Comment (xiexinch) [inline configs/segnext/segnext.yml:17]: Backlog: refine the result table in the README.

File: configs/segnext/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
+# SegNeXt
+
+[SegNeXt: Rethinking Convolutional Attention Design for Semantic Segmentation](https://arxiv.org/abs/2209.08575)
```
Comment (xiexinch) [inline configs/segnext/README.md:3]:
```suggestion
> [SegNeXt: Rethinking Convolutional Attention Design for Semantic Segmentation](https://arxiv.org/abs/2209.08575)
```

File: configs/segnext/README.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+## Introduction
+
+<!-- [ALGORITHM] -->
+
+<a href="https://github.com/visual-attention-network/segnext">Official Repo</a>
+
+<a href="https://github.com/open-mmlab/mmsegmentation/blob/v0.31.0/mmseg/models/backbones/mscan.py#L328">Code Snippet</a>
```
Comment (xiexinch) [inline configs/segnext/README.md:11]:
```suggestion
<a href="https://github.com/open-mmlab/mmsegmentation/blob/dev-1.x/mmseg/models/backbones/mscan.py#L328">Code Snippet</a>
```

**Review by xiexinch — APPROVED** [review]

File: configs/segnext/README.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+- When we integrated SegNeXt into MMSegmentation, we modified some layers' names to make them more precise and concise without changing the model architecture. Therefore, the keys of pre-trained weights are different from the [original weights](https://cloud.tsinghua.edu.cn/d/c15b25a6745946618462/), but don't worry about these changes. we have converted them and uploaded the checkpoints, you might find URL of pre-trained checkpoints in config files and can use them directly for training.
+
+- The total batch size is 16. We trained for SegNeXt with a single GPU as the performance degrades significantly when using`SyncBN` (mainly in `OverlapPatchEmbed` modules of `MSCAN`) of PyTorch 1.9.
+
+- There will be subtle differences when model testing as Non-negative Matrix Factorization (NMF) in `LightHamHead` will be initialized randomly. To control this randomness, please set the random seed when model testing. You can modify [`./tools/test.py`](https://github.com/open-mmlab/mmsegmentation/blob/master/tools/test.py) like:
+
+```python
+def main():
+    from mmseg.apis import set_random_seed
```
Comment (MeowZheng) [inline configs/segnext/README.md:46]: `set_random_seed` has been removed

**Review by MeowZheng — COMMENTED** [review]
Please solve the conflict.

**Comment by anilantony** [comment]
Please help to run SegNeXt in MMSegmentation 2.0 in colab...

**Pull request merged**